src/main/java/com/jcabi/http/wire/LastModifiedCachingWire.java

```diff
@@ -0,0 +1,207 @@
+
+    /**
+     * Last-Modified header name.
+     */
+    public static final String LAST_MODIFIED = "Last-Modified";
+
+    /**
+     * If-Modified-Since header name.
+     */
+    private static final String IF_MODIFIED_SINCE = "If-Modified-Since";
+
+    /**
+     * Cache.
+     */
+    private final transient Map<Request, Response> cache;
+
+    /**
+     * Original wire.
+     */
+    private final transient Wire origin;
+
+    /**
+     * Public ctor.
+     * @param wire Original wire
+     */
+    public LastModifiedCachingWire(final Wire wire) {
+        this.origin = wire;
+        this.cache = new ConcurrentHashMap<>();
+    }
+
+    // @checkstyle ParameterNumber (5 lines)
+    @Override
+    public Response send(final Request req, final String home,
+        final String method,
+        final Collection<Map.Entry<String, String>> headers,
+        final InputStream content,
+        final int connect,
+        final int read) throws IOException {
+        final Response rsp;
+        if (method.equals(Request.GET)) {
+            rsp = this.lookInCache(
+                req, home, method, headers, content, connect, read
+            );
+        } else {
+            rsp = this.origin.send(
+                req, home, method, headers, content, connect, read
+            );
+        }
+        return rsp;
+    }
+
+    /**
+     * Check cache and update if needed.
+     *
+     * @param req Request
+     * @param home URI to fetch
+     * @param method HTTP method
+     * @param headers Headers
+     * @param content HTTP body
+     * @param connect The connect timeout
+     * @param read The read timeout
+     * @return Response obtained
+     * @throws IOException if fails
+     * @checkstyle ParameterNumber (6 lines)
+     */
+    private Response lookInCache(final Request req,
+        final String home,
+        final String method,
+        final Collection<Map.Entry<String, String>> headers,
+        final InputStream content,
+        final int connect,
+        final int read) throws IOException {
+        final Response rspReceived;
+        if (this.cache.containsKey(req)) {
+            rspReceived = this.updateCache(
+                req, home, method, headers, content, connect, read
+            );
+        } else {
+            rspReceived = this.origin.send(
+                req, home, method, headers, content, connect, read
+            );
+            this.addToCache(req, rspReceived);
+        }
+        return rspReceived;
+    }
+
+    /**
+     * Check response and update cache if needed.
+     *
+     * @param req Request
+     * @param home URI to fetch
+     * @param method HTTP method
+     * @param headers Headers
+     * @param content HTTP body
+     * @param connect The connect timeout
+     * @param read The read timeout
+     * @return Response obtained
+     * @throws IOException if fails
+     * @checkstyle ParameterNumber (8 lines)
+     */
+    private Response updateCache(final Request req,
```

Comment:
I would perhaps rename this `validateCacheWithServer`, to use terminology from the HTTP spec.